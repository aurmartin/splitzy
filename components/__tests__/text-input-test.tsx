import * as React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import TextInput from "../text-input";

describe("TextInput", () => {
  it("renders correctly", () => {
    const tree = render(<TextInput placeholder="Enter text" />);
    expect(tree).toMatchSnapshot();
  });

  it("renders with label when provided", () => {
    const { getByText } = render(
      <TextInput label="Test Label" placeholder="Enter text" />,
    );

    expect(getByText("Test Label")).toBeTruthy();
  });

  it("handles text changes correctly", () => {
    const mockOnChangeText = jest.fn();
    const { getByPlaceholderText } = render(
      <TextInput placeholder="Enter text" onChangeText={mockOnChangeText} />,
    );

    const input = getByPlaceholderText("Enter text");
    fireEvent.changeText(input, "Hello World");

    expect(mockOnChangeText).toHaveBeenCalledWith("Hello World");
  });
});
