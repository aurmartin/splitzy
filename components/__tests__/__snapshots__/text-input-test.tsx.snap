// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<TextInput
  placeholder="Enter text"
  placeholderTextColor="#666"
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderColor": "hsl(0 0% 70%)",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 45,
        "padding": 10,
      },
      undefined,
    ]
  }
/>
`;
